### PR TITLE
Added faker.js and moved it to the data factory

### PR DESCRIPTION
**Pull request by MKobylianskyi**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -11,6 +11,7 @@
   "author": "Mykhailo",
   "license": "ISC",
   "devDependencies": {
+    "@faker-js/faker": "7.1.0",
```
Comment (tokenvolt) [inline package.json:14]: should be under dependencies, because we will need it in production mode

File: utils/data-factory.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import { faker } from '@faker-js/faker'
+
+export function generateRandomDataBYOE(byoeData: object) {
+  byoeData['fisrtName'] = faker.name.firstName()
```
Comment (tokenvolt) [inline utils/data-factory.ts:4]: just return a new object instead of mutating the function argument
```ts
return {
  firstName: faker.name.firstName(),
  lastName: faker.name.lastName(),
  // etc
```

File: page-objects/project-pages/ByoePage.ts [hunk 3]
```diff
@@ -73,61 +73,93 @@ export class ByoePage extends BasePage {
   }
 
   async fillForm(
```
Comment (tokenvolt) [inline page-objects/project-pages/ByoePage.ts:75]: I would change the signature to have 1 arg and a custom `Input` type which would include `obj` as well
```ts
type Input = {
  firstName: string
  lastName: string
  // etc
}

fillForm(input: Input) => Promise<void>
```

File: tests/BYOE/byoe-adding.spec.ts [hunk 4]
```diff
@@ -3,27 +3,43 @@ import { ByoePage } from '../../page-objects/project-pages/ByoePage'
...
+    firstName: string
+    lastName: string
+    jobTitle: string
+    companyName: string
+    phoneNumber: string
+    rate: string
+    tag: string
+    country: string
+    timeZone: string
+    object: {
```
Comment (tokenvolt) [inline tests/BYOE/byoe-adding.spec.ts:19]: `object` does not give additional semantics here, so it is reasonable to remove this nesting level.

File: tests/BYOE/byoe-adding.spec.ts [hunk 5]
```diff
@@ -3,27 +3,43 @@ import { ByoePage } from '../../page-objects/project-pages/ByoePage'
 import { LoginPage } from '../../page-objects/LoginPage'
 import { ExpertsPage } from '../../page-objects/project-pages/ExpertsPage'
 import { getRandomString } from '../../utils/data-helpers'
-import { ProjectsPage } from '../../page-objects/ProjectsPage'
+import { generateRandomDataBYOE } from '../../utils/data-factory'
 
 test.describe('BYOE Adding feature', () => {
+  type Input = {
```
Comment (tokenvolt) [inline tests/BYOE/byoe-adding.spec.ts:9]: please move type definition to a module level outside of a `test` function

File: tests/BYOE/byoe-editing.spec.ts [hunk 6]
```diff
@@ -3,24 +3,41 @@ import { ByoePage } from '../../page-objects/project-pages/ByoePage'
 import { LoginPage } from '../../page-objects/LoginPage'
 import { getRandomString } from '../../utils/data-helpers'
 import { ExpertsPage } from '../../page-objects/project-pages/ExpertsPage'
+import { generateRandomDataBYOE } from '../../utils/data-factory'
 
 test.describe('BYOE Editing feature', () => {
+  type Input = {
```
Comment (tokenvolt) [inline tests/BYOE/byoe-editing.spec.ts:9]: same about moving to the module level

File: utils/data-factory.ts [hunk 7]
```diff
@@ -0,0 +1,21 @@
+import { faker } from '@faker-js/faker'
+
+export function generateRandomDataBYOE(env: number) {
+  const fs = require('fs')
+  let rawdata = fs.readFileSync('test-data/byoe-data.json')
```
Comment (tokenvolt) [inline utils/data-factory.ts:5]: is there any reason we need to store additional data in a separate json file?

Comment (MKobylianskyi) [reply utils/data-factory.ts:5]: Yes, as following data are not generic and should be specified. 
 ```
emailpart, sourceOption, currencyOptionIndex, angleOptionIndex, linkedinURl
```
I.E. 
emailpart - contain email address of the engineer  that supposed  to get emails behalf of the Expert (for which profile we use email)  same as LinkedinURL
sourceOption, currencyOptionIndex, angleOptionIndex -  values that can't be generic and depends on project context. 

For now, it's useful to take this information  in JSON until Project creation tests would be implemented and used as preconditions for current tests. 

To sum up:   this JSON files temporary solution until we have proper implementation of complex  preconditions

File: tests/BYOE/byoe-editing.spec.ts [hunk 8]
```diff
@@ -3,24 +3,40 @@ import { ByoePage } from '../../page-objects/project-pages/ByoePage'
...
   let loginPage: LoginPage
   let expertsPage: ExpertsPage
   const fs = require('fs')
-  let rawdata = fs.readFileSync('test-data/byoe-data.json')
-  const byoeList = JSON.parse(rawdata)
-  //Specify BYOE data set
-  let BYOE = byoeList[1]
-  //Specify BYOE data set
-  rawdata = fs.readFileSync('test-data/env-data.json')
+  let rawdata = fs.readFileSync('test-data/env-data.json')
```
Comment (tokenvolt) [inline tests/BYOE/byoe-editing.spec.ts:31]: 1. Reading files is an expensive operation. It is better to read them once and store in a variable for future reference.
2. You don't need `fs` to read a JSON file. `const data = require("test-data/env-data.json")` should do the trick.

Comment (tokenvolt) [reply tests/BYOE/byoe-editing.spec.ts:31]: 3. It is common practice to put require at the beginning of the file.
